d/source/flexver.d

```diff
@@ -0,0 +1,440 @@
+///
+module flexver;
+
+import std.ascii: isDigit;
+import std.uni: byCodePoint;
+
+///
+int compare(in string first, in string second) pure @safe
+{
+	auto decomposedFirst = decompose(first);
+	auto decomposedSecond = decompose(second);
+
+	int componentIndex;
```

Comment:
This could be an index in the foreach instead of manually tracking

Reply:
it's needed outside of the foreach, line 35